distributed/s3.py

```diff
@@ -0,0 +1,34 @@
+import boto3
+from dask.imperative import Value
+from distributed import default_executor
+from toolz import curry
+
+DEFAULT_PAGE_LENGHT = 1000
+
+
+def get_list_of_summary_objects(bucket_name, prefix='', page_size=DEFAULT_PAGE_LENGHT):
+    s3 = boto3.resource('s3')
+    return [summary_object for summary_object in
+            s3.Bucket(bucket_name).objects.filter(Prefix=prefix).page_size(page_size)]
+
+
+def read_content_from_keys(bucket, key):
+    import boto3
+    s3 = boto3.resource('s3')
+    return s3.Object(bucket, key).get()['Body'].read()
+
+
+def read_contents(bucket_name, executor=None, prefix='', lazy=False):
```

Comment:
Can we move the prefix keyword argument to be second?  I imagine people will more often use it than the executor keyword argument and it would be nice to support it as an unnamed input like the following:

    s3.read_contents('mybucket', '/data')